src/common/components/table/Table.tsx

```diff
@@ -288,8 +293,18 @@ export function Table<T>({
                 <ChakraTable.ColumnHeader
                   key={header.id}
                   css={{ ...getCommonPinningStyles(column) }}
-                  py={3}
-                  px={[2, 2, 2, `clamp(12px, calc(48px / ${columns.length}), 16px)`]}
+                  py={[
+                    2,
+                    2,
+                    2,
+                    `clamp(var(--stacks-spacing-3), calc(var(--stacks-spacing-12) / ${columns.length}), var(--stacks-spacing-4))`,
```

Comment:
If we use the same clamp value for both px and py, we might want to define it as a constant in the same file. Also consider building it from smaller more readable constants like TABLE_CELL_MIN_PADDING, TABLE_CELL_MAX_PADDING, and TABLE_CELL_BASE_SPACING.